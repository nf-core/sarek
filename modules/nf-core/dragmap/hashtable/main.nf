process DRAGMAP_HASHTABLE {
    tag "$fasta"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/dragmap:1.2.1--h72d16da_1':
        'biocontainers/dragmap:1.2.1--h72d16da_1' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("dragmap")    , emit: hashmap
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    mkdir dragmap
    dragen-os \\
        --build-hash-table true \\
        --ht-reference $fasta \\
        --output-directory dragmap \\
        $args \\
        --ht-num-threads $task.cpus

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dragmap: \$(echo \$(dragen-os --version 2>&1))
    END_VERSIONS
    """

    stub:
    """
    mkdir dragmap

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dragmap: \$(echo \$(dragen-os --version 2>&1))
    END_VERSIONS
    """

}
