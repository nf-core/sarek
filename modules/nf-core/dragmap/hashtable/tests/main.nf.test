nextflow_process {

    name "Test Process DRAGMAP_HASHTABLE"
    script "../main.nf"
    process "DRAGMAP_HASHTABLE"
    tag "modules"
    tag "modules_nfcore"
    tag "dragmap"
    tag "dragmap/hashtable"

    test("sarscov2 - fasta") {

        when {

            process {
                """
                input[0] = [
                    [id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                        file(process.out.hashmap[0][1]).name,
                        file(process.out.versions[0]).name
                    ).match()
                }
            )
        }

    }

    test("sarscov2 - fasta - stub") {

        options "-stub"
        when {

            process {
                """
                input[0] = [
                    [id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                        file(process.out.hashmap[0][1]).name,
                        file(process.out.versions[0]).name
                    ).match()
                }
            )
        }

    }

    // TODO Add test using alt-masked bed file
    // https://github.com/Illumina/dragmap#build-hash-table-using-an-alt-masked-bed-file

}
