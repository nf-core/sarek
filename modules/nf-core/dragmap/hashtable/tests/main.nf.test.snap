{
    "sarscov2 - fasta - stub": {
        "content": [
            "dragmap",
            "versions.yml"
        ],
        "timestamp": "2023-11-22T13:42:22.53412378"
    },
    "sarscov2 - fasta": {
        "content": [
            "dragmap",
            "versions.yml"
        ],
        "timestamp": "2023-11-22T12:53:25.493202451"
    }
}